### PR TITLE
Large tests for schedule command trigger

### PR DESCRIPTION
**Pull request by dairyo**
* Move PostNewTriggerForSchedulePredicateTests.swift to ThingIFAPIScheduleTriggerTests.swift
* Fix some bugs. I followed implementation of thing-if AndroidSDK for below fixes.
  * `Command.commandID` becomes optional.
  * `Command.commandState` becomes optional.
  * `Command.created` becomes optional.

Related PR: #296

### CODE REVIEW

**Review by naoki-akimoto — APPROVED** [review]

File: ThingIFSDK/ThingIFSDK/Command.swift [hunk 1]
```diff
@@ -128,6 +125,20 @@ extension Command: FromJsonObject {
...
+            commandState = CommandState(rawValue: state)
+        } else {
+            commandState = nil
+        }
+
+        let created: Date?
+        if let date = jsonObject["created"] as? Int64 {
+            created = Date(timeIntervalSince1970InMillis: date)
+        } else {
+            created = nil
```
Comment (rizakii) [inline ThingIFSDK/ThingIFSDK/Command.swift:139]: remove this, its not needed.

File: ThingIFSDK/ThingIFSDKLargeTests/ThingIFAPIScheduleTriggerTests.swift [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            XCTAssertNil(error)
+        }
+
+        expectation = self.expectation(description: "list trigger second")
+        self.onboardedApi.listTriggers(100) { triggers, paginationKey, error in
+
+            { () in
+                XCTAssertNil(error)
+                XCTAssertNil(paginationKey)
+                guard let triggers = triggers else {
```
Comment (rizakii) [inline ThingIFSDK/ThingIFSDKLargeTests/ThingIFAPIScheduleTriggerTests.swift]: no need to use guard, use `XCTAssertNotNil`

**Review by yongpingchen — APPROVED** [review]

File: ThingIFSDK/ThingIFSDKTests/CommandTests.swift [hunk 3]
```diff
@@ -77,7 +77,7 @@ class CommandTests: SmallTestBase {
         XCTAssertEqual(issuerID, actual.issuerID)
         XCTAssertEqual(aliasActions, actual.aliasActions)
         XCTAssertEqual([], actual.aliasActionResults)
-        XCTAssertEqual(.sending, actual.commandState)
+        XCTAssertEqual(nil, actual.commandState)
```
Comment (rizakii) [inline ThingIFSDK/ThingIFSDKTests/CommandTests.swift]: use assertNil

File: ThingIFSDK/ThingIFSDKTests/ThingIFAPIGetCommandTests.swift [hunk 4]
```diff
@@ -173,7 +173,7 @@ class ThingIFAPIGetCommandTests: SmallTestBase {
             nil)
         iotSession = MockSession.self
 
-        setting.api.getCommand(testCase.commandID) { command, error -> Void in
+        setting.api.getCommand(testCase.commandID!) { command, error -> Void in
```
Comment (rizakii) [inline ThingIFSDK/ThingIFSDKTests/ThingIFAPIGetCommandTests.swift]: remove `-> Void `

**Review by rizakii — CHANGES_REQUESTED** [review]

File: ThingIFSDK/ThingIFSDK/Command.swift (hunk 1 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/Command.swift:139]: @rizakii 

This is required.
We must initialize `let` value before using it. If we remove `else` statement, it cause build error.

File: ThingIFSDK/ThingIFSDKLargeTests/ThingIFAPIScheduleTriggerTests.swift (hunk 2 above)
Comment (dairyo) [reply ThingIFSDK/ThingIFSDKLargeTests/ThingIFAPIScheduleTriggerTests.swift]: @rizakii 

I want to use `triggers` as non optional so I used `guard`. `XCTAssertNotNil` does not unwrap optional value.

Comment (rizakii) [reply ThingIFSDK/ThingIFSDKLargeTests/ThingIFAPIScheduleTriggerTests.swift]:
```swift
XCTAssertNotNil(triggers)
let triggers = triggers!
```
better than use `guard`

**Review by rizakii — APPROVED** [review]

**Comment by rizakii** [comment]
LGTM

**Pull request merged**